File: snippets/go/snippet-03.go
package main

import (
  "context"
  "fmt"
  "strings"

  apiclientgo "github.com/gleanwork/api-client-go"
  "github.com/gleanwork/api-client-go/models/components"
)

// chatAnswer concatenates the text fragments of a chat response.
func chatAnswer(resp *components.ChatResponse) string {
  var sb strings.Builder
  for _, msg := range resp.Messages {
    for _, frag := range msg.Fragments {
      if frag.Text != nil {
        sb.WriteString(*frag.Text)
      }
    }
  }
  return sb.String()
}

func chatExample(s *apiclientgo.Glean) error {
  ctx := context.Background()

  res, err := s.Client.Chat.Create(ctx, components.ChatRequest{
    Messages: []components.ChatMessage{
      {
        Fragments: []components.ChatMessageFragment{
          {Text: apiclientgo.Pointer("Explain our Q4 strategy")},
        },
      },
    },
  }, nil, nil)
  if err != nil {
    return err
  }

  if res.ChatResponse != nil {
    fmt.Println(chatAnswer(res.ChatResponse))
  }
  return nil
}

File: snippets/java/snippet-03.java
public class ChatService {
  private final Glean client;
  
  public ChatService(String apiToken, String serverURL) {
    this.client = Glean.builder()
        .apiToken(apiToken)
        .serverURL(serverURL)
        .build();
  }
  
  public String sendMessage(String message) throws Exception {
    var response = client.client().chat().create()
        .chatRequest(ChatRequest.builder()
            .messages(List.of(
                ChatMessage.builder()
                    .fragments(List.of(
                        ChatMessageFragment.builder()
                            .text(message)
                            .build()))
                    .build()))
            .build())
        .call();
        
    return response.chatResponse()
        .map(chat -> chat.messages().stream()
            .flatMap(msg -> msg.fragments().stream())
            .map(fragment -> fragment.text().orElse(""))
            .collect(Collectors.joining()))
        .orElse("No response");
  }
}
